### DISCUSSION TITLE
Enabling SSL for non-cloud(on premise) not working

### DISCUSSION THREAD

**Discussion by dinchand** [ing-bank/cassandra-jdbc-wrapper#34] [Q&A]
I am trying to use the driver to run liquibase for cassandra on our on-premise . However the `enableSsl` is dropped if I use a `configFile`. Also if I do not use a config file the I see the following in the code
```
 // SSL configuration.
        if (StringUtils.isBlank(cloudSecureConnectBundle)) {
            if (sslEnabled) {
                if (StringUtils.isNotEmpty(sslEngineFactoryClassName)) {
                    configureSslEngineFactory(builder, sslEngineFactoryClassName);
                } else {
                    driverConfigLoaderBuilder.withBoolean(DefaultDriverOption.SSL_HOSTNAME_VALIDATION,
                        sslHostnameVerificationEnabled);
                    configureDefaultSslEngineFactory(builder, driverConfigLoaderBuilder);
                }
            }
        } else {
            LOG.info("Cloud secure connect bundle used. SSL will always be enabled. All manual SSL "
                + "configuration(s) will be ignored.");
        }
```

So, is my understanding right that the case for SSL for non-cloud is not supported yet ?

**Comment by maximevw** [comment]
Hello @dinchand,

SSL should be supported for both cloud and non-cloud connections:

- For non-cloud connections:
  - if you use a configuration file through the parameter `configfile` (as explained [here](https://github.com/ing-bank/cassandra-jdbc-wrapper#using-a-configuration-file)), other SSL parameters are indeed ignored and you have to ensure the Cassandra client configuration file contains the appropriate configuration in the section `advanced.ssl-engine-factory` (see more details in the [official documentation](https://docs.datastax.com/en/developer/java-driver/latest/manual/core/configuration/reference/))
  - if you're not using a configuration file, you have to enable SSL with the parameter `enablessl` and either set JSSE configuration in your environment (you can look at [here](https://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/JSSERefGuide.html#how-to-specify-a-java-lang-system-property)) or configure a custom `SslEngineFactory` with `sslenginefactory` parameter as explained [here](https://github.com/ing-bank/cassandra-jdbc-wrapper#secure-connection-with-ssl) (not recommended)
- For cloud connections, the SSL configuration is included in the secure connect bundle, that's why SSL parameters from the JDBC connection string are ignored in this case

If you still have issues after verifying your configuration file is correct, do not hesitate to open an issue with details enough to investigate (JDBC URL you use, your configuration file (at least the SSL configuration part - redacted possible sensitive data), error(s) you get, ...).
